rust/src/core.rs

```diff
@@ -39,6 +39,12 @@ pub const STREAM_GAP:      u8 = 0x10;
 pub const STREAM_DEPTH:    u8 = 0x20;
 pub const STREAM_MIDSTREAM:u8 = 0x40;
 
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Direction {
+    ToServer = 0x04,
+    ToClient = 0x08,
+ }
+
```

Comment:
Can you try adding repr(C) here as well?  Then maybe it can be the source of truth and used from C as well.